installer/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/SaaSBoostInstall.java

```diff
@@ -959,15 +1063,65 @@ protected void deleteApplicationConfig() {
     }
 
     protected void deleteProvisionedTenant(LinkedHashMap<String, Object> tenant) {
-        // Arguably, the delete tenant API should be called here, but all it currently does is set the tenant status
-        // to disabled and then fire off an async event to delete the CloudFormation stack. We could do that, but we'd
-        // still need to either subscribe to the SNS topic for DELETE_COMPLETE notification or we'd need to cycle on
-        // describe stacks like we do elsewhere to wait for CloudFormation to finish. In either case, we need to know
-        // the stack name for the tenant.
-        // Also, this won't scale super well and could be parallelized...
-        String tenantStackName = "Tenant-" + ((String) tenant.get("id")).split("-")[0];
-        outputMessage("Deleteing tenant CloudFormation stack " + tenantStackName);
-        deleteCloudFormationStack(tenantStackName);
+        // TODO we can parallelize to improve performance with lots of tenants
+        String tenantStackId = (String) ((HashMap<String, Object>)((HashMap<String, Object>) tenant.get("resources")).get("CLOUDFORMATION")).get("arn");
+        final ObjectMapper mapper = new ObjectMapper();
+        try {
+            Map<String, Object> systemApiRequest = new HashMap<>();
+            Map<String, Object> detail = new HashMap<>();
+            detail.put("resource", "tenants/" + (String) tenant.get("id"));
+            detail.put("method", "DELETE");
+            String tenantId = (String) tenant.get("id");
+            Map<String, String> tenantIdOnly = new HashMap<>();
+            tenantIdOnly.put("id", tenantId);
+            detail.put("body", mapper.writeValueAsString(tenantIdOnly));
+            systemApiRequest.put("detail", detail);
+            final byte[] payload = mapper.writeValueAsBytes(systemApiRequest);
+            try {
+                LOGGER.info("Invoking delete tenant API");
+                InvokeResponse response = lambda.invoke(request -> request
+                        .functionName("sb-" + this.envName + "-private-api-client")
+                        .invocationType(InvocationType.REQUEST_RESPONSE)
+                        .payload(SdkBytes.fromByteArray(payload))
+                );
+                if (response.sdkHttpResponse().isSuccessful()) {
+                    LOGGER.debug("got response back: {}", response);
+                    // wait for tenant to reach deleted
+                    final String DELETED = "deleted";
+                    LocalDateTime timeout = LocalDateTime.now().plus(30, ChronoUnit.MINUTES);
```

Comment:
Let's make this timeout be 60 minutes -- remember we're waiting for CloudFormation to delete multiple stacks and potentially are dealing with RDS instances and FSx file systems which can take a very long time to delete.